### PR TITLE
added: ability to open the smartplaylisteditor for a given playlist

### PR DESCRIPTION
**Pull request by Montellese**
This is a copy of @notspiff's work from #6966 and allows to open the smartplaylist editor for a specific smartplaylist through builtins.

### CODE REVIEW

File: xbmc/dialogs/GUIDialogSmartPlaylistEditor.cpp [hunk 1]
```diff
@@ -31,6 +31,7 @@
 #include "FileItem.h"
 #include "input/Key.h"
 #include "guilib/LocalizeStrings.h"
+#include "utils/log.h"
```
Comment (garbear) [inline xbmc/dialogs/GUIDialogSmartPlaylistEditor.cpp]: left over from testing? i don't see CLog in use in this file

Comment (Montellese) [inline xbmc/dialogs/GUIDialogSmartPlaylistEditor.cpp]: You're right. Thanks.

**Comment by Montellese** [comment]
jenkins build this please

**Comment by Montellese** [comment]
Only some binary addons that failed to build.

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by Lunatixz** [comment]
Can someone explain how this works? 
ie.
xbmc.executebuiltin( "ActivateWindow(10136,%s)" % xbmc.translatePath(smartplaylist.xsp) )

**Comment by Lunatixz** [comment]
Thanks for adding this to Jarvis... One small problem, if you call xbmc.executebuiltin( "ActivateWindow(10136,%s)" % xbmc.translatePath(smartplaylist.xsp) ) twice, the previous playlist rules are still displayed in the editor, but are not written to the xsp. Only the "find items where" window keeps remanent info, all other properties change between playlists.

**Comment by phil65** [comment]
We´re aware of it: https://github.com/xbmc/xbmc/pull/9545
Didnt find the time to look into it again yet though.

**Pull request merged**